torchdata/datapipes/iter/util/dataframemaker.py

```diff
@@ -39,3 +41,40 @@ def __new__(
         batch_dp = source_dp.batch(dataframe_size)
         df_dp = batch_dp.map(partial(torcharrow.DataFrame, dtype=dtype, columns=columns, device=device))
         return df_dp
+
+
+@functional_datapipe("load_parquet_as_df")
+class ParquetDFIterDataPipe(IterDataPipe):  # IterDataPipe[torcharrow.IDataFrame[T_co]]
+    r"""
+    Iterable DataPipe that takes in paths to Parquet files and return a TorchArrow DataFrame for each row group
+    within a Parquet file.
+
+    Args:
+        source_dp: source DataPipe containing paths to the Parquet files
+        columns: List of str that specifies the column names of the DataFrame
+        use_threads: if True, Parquet reader will perform multi-threaded column reads
+        dtype: specify the TorchArrow dtype for the DataFrame
+        device: specify the device on which the DataFrame will be stored
```

Comment:
The API should be same as the `DataFrameMaker`.
- arguments order
- `dataframe_size`

`use_threads` can be a kwargs at the end.

Reply:
I updated the argument order but `dataframe_size` may not be exactly relevant here. In `DataFrameMaker` we have can read and save one row at a time, but for `parquet`, we read one row group at a time (which is many rows as defined by the author of the specific dataset).